tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -395,9 +398,353 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+		ginkgo.By(fmt.Sprintf("Creating a 512mb test data file %v", testdataFile))
+		op, err := exec.Command("dd", "if=/dev/urandom", fmt.Sprintf("of=%v", testdataFile),
+			"bs=64k", "count=8000").Output()
+		fmt.Println(op)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			op, err = exec.Command("rm", "-f", testdataFile).Output()
+			fmt.Println(op)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expand pvcs while writing some data on them")
+		var wg sync.WaitGroup
+		currentPvcSize := pvcs[0].Spec.Resources.Requests[v1.ResourceStorage]
+		newSize := currentPvcSize.DeepCopy()
+		newSize.Add(resource.MustParse("2Gi"))
+		fsSizes := []int64{}
+		for _, pod := range pods {
+			originalSizeInMb, err := getFSSizeMb(f, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			fsSizes = append(fsSizes, originalSizeInMb)
+		}
+		wg.Add(4)
+		go copyFile2PodInParallel(pods[0], testdataFile, &wg)
+		go copyFile2PodInParallel(pods[1], testdataFile, &wg)
+		go resize(client, pvcs[0], pvcs[0].Spec.Resources.Requests[v1.ResourceStorage], newSize, &wg)
+		go resize(client, pvcs[1], pvcs[1].Spec.Resources.Requests[v1.ResourceStorage], newSize, &wg)
+		wg.Wait()
+
+		ginkgo.By("Wait and verify the file system resize on pvcs")
+		for i := range pvcs {
+			framework.Logf("Waiting for file system resize to finish for pvc %v", pvcs[i].Name)
+			pvcs[i], err = waitForFSResize(pvcs[i], client)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			pvcConditions := pvcs[i].Status.Conditions
+			expectEqual(len(pvcConditions), 0, "pvc %v should not have status conditions", pvcs[i].Name)
+
+			var fsSize int64
+			framework.Logf("Verify filesystem size for mount point /mnt/volume1 for pod %v", pods[i].Name)
+			fsSize, err = getFSSizeMb(f, pods[i])
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			framework.Logf("File system size after expansion : %v", fsSize)
+			gomega.Expect(fsSize > fsSizes[i]).To(gomega.BeTrue(),
+				fmt.Sprintf(
+					"filesystem size %v is not > than before expansion %v for pvc %q",
+					fsSize, pvcs[i].Name, fsSizes[i]))
+
+			framework.Logf("File system resize finished successfully")
+		}
+		ginkgo.By("Verify that the expanded CNS volumes are compliant and have correct policy id")
+		for i, pv := range pvs {
+			volumeID := pv.Spec.CSI.VolumeHandle
+			storagePolicyMatches, err := e2eVSphere.VerifySpbmPolicyOfVolume(volumeID, policyNames[i])
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(storagePolicyMatches).To(gomega.BeTrue(), "storage policy verification failed")
+			e2eVSphere.verifyVolumeCompliance(volumeID, true)
+		}
+
+		ginkgo.By("Verify the data on the PVCs match what was written in step 7")
+		for _, pod := range pods {
+			_ = framework.RunKubectlOrDie(namespace, "cp",
+				fmt.Sprintf("%v/%v:/mnt/volume1/testdata", namespace, pod.Name), testdataFile+pod.Name)
+			defer func() {
+				op, err = exec.Command("rm", "-f", testdataFile+pod.Name).Output()
```

Comment:
I think we can put these "cp", "rm" etc. as constants as these are being used in many places

Reply:
Can you raise a separate automation bug for this. I don't want to test all the cases which have such code for this PR.